enzyme/test/TypeAnalysis/smax.ll

```diff
@@ -0,0 +1,24 @@
+; RUN: if [ %llvmver -lt 16 ]; then %opt < %s %loadEnzyme -print-type-analysis -type-analysis-func=smax -o /dev/null | FileCheck %s; fi
+; RUN: %opt < %s %newLoadEnzyme -passes="print-type-analysis" -type-analysis-func=smax -S -o /dev/null | FileCheck %s
+
+define i32 @smax(i32 %a, i32 %b) {
+entry:
+  %0 = call i32 @llvm.smax.i32(i32 %a, i32 %b)
+  %1 = add i32 %a, 1
```

Comment:
You need this to be an unknown value (eg from a random call), since otherwise it will already be known as an int from the argument.

Otherwise this won't actually test the propagation